client/src/app/components/transaction-history/transaction-history.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, OnInit, Input, OnChanges } from '@angular/core';
+import { Transaction } from 'src/app/interfaces/transaction';
+
+@Component({
+  selector: 'app-transaction-history',
+  templateUrl: './transaction-history.component.html',
+  styleUrls: ['./transaction-history.component.scss']
+})
+export class TransactionHistoryComponent implements OnInit, OnChanges {
+  @Input() userTransactions: Transaction[];
+  transactionHistory: Transaction[];
+  constructor() { }
+
+  ngOnInit(): void {
+
+  }
+  ngOnChanges(): void {
+    if (this.userTransactions) { this.transactionHistory = this.userTransactions; }
+    console.log('Transaction history: ', this.transactionHistory);
+  }
+  transactionFormatStart(transaction: Transaction): string {
+    let printedTransaction = '';
+    if (transaction) {
+      transaction.type === 'MARKET_BUY' ? printedTransaction += 'Bought ' : printedTransaction += 'Sold ';
```

Comment:
this line can be re-written as below. It does the same thing, it's just more readable

```printedTransaction += transaction.type === 'MARKET_BUY' ?  'Bought ' : 'Sold ';```